src/pages/[platform]/build-a-backend/data/custom-business-logic/search-and-aggregate-queries/index.mdx

```diff
@@ -0,0 +1,369 @@
+  {
+    blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+    bucketName: "opensearch-backup-bucket-amplify-gen-2-test",
+    enforceSSL: true,
+    versioned: true,
+    autoDeleteObjects: true,
+    removalPolicy: RemovalPolicy.DESTROY,
+  }
+);
+
+const openSearchIntegrationPipelineRole = new iam.Role(
+  openSearchStack,
+  "OpenSearchIntegrationPipelineRole",
+  {
+    assumedBy: new iam.ServicePrincipal("osis-pipelines.amazonaws.com"),
+    inlinePolicies: {
+      openSearchPipelinePolicy: new iam.PolicyDocument({
+        statements: [
+          new iam.PolicyStatement({
+            actions: ["es:DescribeDomain"],
+            resources: [openSearchDomain.domainArn],
+            effect: iam.Effect.ALLOW,
+          }),
+          new iam.PolicyStatement({
+            actions: ["es:ESHttp*"],
+            resources: [openSearchDomain.domainArn],
+            effect: iam.Effect.ALLOW,
+          }),
+        ],
+      }),
+    },
+    managedPolicies: [
+      iam.ManagedPolicy.fromAwsManagedPolicyName(
+        "AmazonOpenSearchServiceFullAccess"
+      ),
+    ],
+  }
+);
+
+openSearchIntegrationPipelineRole.addManagedPolicy(
+  iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonDynamoDBFullAccess")
+);
+openSearchIntegrationPipelineRole.addManagedPolicy(
+  iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonS3FullAccess")
+);
+
+```
+For the S3 bucket, follow standard security practices: block public access, encrypt data at rest, and enable versioning.
+
+The IAM role should allow the OpenSearch Ingestion Service (OSIS) pipelines to assume it. Grant specific OpenSearch Service permissions and also provide DynamoDB and S3 access. You may customize permissions to follow the principle of least privilege.
+
+### Step 2b: OpenSearch Service Pipeline
+
+Define the pipeline construct and its configuration.
+
+The configuration is a data-prepper feature of OpenSearch. For specific documentation on DynamoDB configuration, refer to [OpenSearch documentation](https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sources/dynamo-db/).
+
+``` ts title="amplify/backend.ts"
+
+const openSearchTemplate = `
+version: "2"
+dynamodb-pipeline:
+  source:
+    dynamodb:
+      acknowledgments: true
+      tables:
+        - table_arn: "${backend.data.resources.tables["Todo"].tableArn}"
+          # Remove the stream block if only export is needed
+          stream:
+            start_position: "LATEST"
+          # Remove the export block if only stream is needed
+          export:
+            s3_bucket: "${s3BackupBucket.bucketName}"
+            s3_region: "us-east-2"
+            s3_prefix: "<tableName>/"
+      aws:
+        sts_role_arn: "${openSearchIntegrationPipelineRole.roleArn}"
+        region: "us-east-2"
+  sink:
+    - opensearch:
+        hosts:
+          [
+            "https://${openSearchDomain.domainEndpoint}",
+          ]
+        index: "${indexName}"
+        index_type: "custom"
```

Comment:
this works well for indexing a single table, what's the best practice for indexing multiple tables? Do you create separate pipelines? Different indexes?

Reply:
added reference link for indexing multiple tables